types/reportattributes.pp

```diff
@@ -0,0 +1,11 @@
+type Servicenow_reporting_integration::ReportAttributes = Variant[Enum['none','all'],
+                                                                  Array[Enum['failed_changes',
+                                                                            'corrective_changes',
+                                                                            'intentional_changes',
+                                                                            'pending_changes',
```

Comment:
I find the fact that we have more fine-grained _concrete_ changes (corrective_changes and intentional_changes) but not-as-fine-grained pending changes a bit weird. I get that the latter's b/c of Puppet's (current) report processor API, but it's still a weird asymmetry. So I'm wondering if we should:

1. Add a `changes` attribute that's a shorthand for both `corrective_changes` and `intentional_changes` OR

2. Exclude `pending_changes` until https://tickets.puppetlabs.com/browse/PUP-10607 is finished, whereby we can then add `pending_intentional_changes`, `pending_corrective_changes`? I do recall that supporting noop reports wasn't important for MVP.

I'm also OK with keeping it as-is since `pending_changes` lets us ship with _some_ noop report functionality. Just a thought.

Reply:
Nevermind ignore me. I'm leaning towards keeping `pending_changes` b/c it lets us ship w/ noop functionality.